contrib/scalar/scalar.c

```diff
@@ -164,7 +164,7 @@ static int set_recommended_config(void)
 	for (i = 0; config[i].key; i++) {
 		char *value;
 
-		if (git_config_get_string(config[i].key, &value)) {
+		if (overwrite || git_config_get_string(config[i].key, &value)) {
```

Comment:
This is an interesting way to approach the case of "we recommended `key=X` before, but now recommend `key=Y`" but I think we should be more careful than just overwriting everything.

If we need to update an established config during an upgrade, we should consider marking the list of key/value pairs with an `overwrite` member, and we can then only force a subset of config options and continue to allow users to change others as necessary.

Reply:
I implemented this in https://github.com/microsoft/git/pull/359/commits/fede3ad2820485fb61e7b221ae3456c9ded56aa7 and went through the list of config settings that I thought make sense to overwrite/override on reconfigure. Could I bother you to review the list, too, and if you have suggestions how to improve it, let me know?